doc/man3/OSSL_CRMF_MSG_set_validity.pod

```diff
@@ -0,0 +1,103 @@
+=pod
+
+=head1 NAME
+
+OSSL_CRMF_MSG_set_validity,
+OSSL_CRMF_MSG_set_certReqId,
+OSSL_CRMF_CERTTEMPLATE_fill,
+OSSL_CRMF_MSG_set0_extensions,
+OSSL_CRMF_MSG_push0_extension,
+OSSL_CRMF_MSG_create_popo,
+OSSL_CRMF_MSGS_verify_popo
+- functions populating and verifying CRMF CertReqMsg structures
+
+=head1 SYNOPSIS
+
+ #include <openssl/crmf.h>
+
+ int OSSL_CRMF_MSG_set_validity(OSSL_CRMF_MSG *crm, time_t from, time_t to);
+
+ int OSSL_CRMF_MSG_set_certReqId(OSSL_CRMF_MSG *crm, const int rid);
+
+ int OSSL_CRMF_CERTTEMPLATE_fill(OSSL_CRMF_CERTTEMPLATE *tmpl,
+                                 const EVP_PKEY *pubkey,
+                                 const X509_NAME *subject,
+                                 const X509_NAME *issuer,
+                                 const ASN1_INTEGER *serial);
+
+ int OSSL_CRMF_MSG_set0_extensions(OSSL_CRMF_MSG *crm,
+                                   X509_EXTENSIONS *exts);
+
+ int OSSL_CRMF_MSG_push0_extension(OSSL_CRMF_MSG *crm,
+                                   const X509_EXTENSION *ext);
+
+ int OSSL_CRMF_MSG_create_popo(OSSL_CRMF_MSG *crm, const EVP_PKEY *pkey,
+                               int dgst, int ppmtd);
+
+ int OSSL_CRMF_MSGS_verify_popo(const OSSL_CRMF_MSGS *reqs,
+                                int rid, int acceptRAVerified);
+
+=head1 DESCRIPTION
+
+OSSL_CRMF_MSG_set_validity() sets B<from> as notBefore and B<to> as notAfter
+as the validity in the certTemplate of B<crm>.
+
+OSSL_CRMF_MSG_set_certReqId() sets B<rid> as the certReqId of B<crm>.
+
+OSSL_CRMF_CERTTEMPLATE_fill() sets those fields of the certTemplate B<tmpl>
+for which non-NULL values are provided: B<pubkey>, B<subject>, B<issuer>,
+and/or B<serial>. The arguments are not consumed.
+
+OSSL_CRMF_MSG_set0_extensions() sets B<exts> as the extensions in the
+certTemplate of B<crm>. Consumes B<exts>.
+
+OSSL_CRMF_MSG_push0_extension() pushes the X509 extension B<ext> to the
+extensions in the certTemplate of B<crm>.  Consumes B<ext>.
+
+OSSL_CRMF_MSG_create_popo() creates and sets the Proof-of-Possession (POP)
```

Comment:
Aehm, sorry, but "popo" is the german word for butt.
Could we please have another name?

Reply:
:-) That's funny!

Actually though it is consistently called popo in the RFC!!

Reply:
> Actually though it is consistently called popo in the RFC!!

Actually they don't use it consistently: I counted only 23 POPOs vs 71-23 = 48 POPs.

```
curl https://www.ietf.org/rfc/rfc4210.txt | tr ' ' '\n' | grep POP
71
curl https://www.ietf.org/rfc/rfc4210.txt | tr ' ' '\n' | grep POPO
23
```
So why not use POP instead of POPO?

Reply:
I think the waned to have a prove that the joke is really funny...

Reply:
Both Martin and me are native German speakers.
We're aware of this accidental pun and got used to it :)

Reply:
Note the following from the RFC's "Changes since RFC 2511" section:

>    8.  pop field in the CertReqMsg structure has been renamed to popo to
>        avoid confusion between POP and pop.

The term "pop" is significantly more [overloaded in English](https://dict.leo.org/englisch-deutsch/pop), than in German. I did a random check; Google Translate doesn't know a special meaning for "popo" for many languages, I only stumbled over the ["mother-in-law" meaning in Chinese](https://translate.google.com/#zh-CN/en/popo).

To ensure that those who are now eager to put their newly acquired German skills into practice do it right: in German the word "Popo" (capitalized in German as it is a noun) is rather a children's term for buttocks, it cannot meaningfully be used anymore when the size of the object in question has grown across a certain subjective threshold.

If nobody feels very strongly about that, I would just stay with the short term explicitly used by the RFC for the relevant ASN.1 field identifier.

Reply:
Note the following from the RFC's "Changes since RFC 2511" section:

>    8.  pop field in the CertReqMsg structure has been renamed to popo to
>        avoid confusion between POP and pop.

The term "pop" is significantly more [overloaded in English](https://dict.leo.org/englisch-deutsch/pop), than in German. I did a random check; Google Translate doesn't know a special meaning for "popo" for many languages, I only stumbled over the ["mother-in-law" meaning in Chinese](https://translate.google.com/#zh-CN/en/popo).

To ensure that those who are now eager to put their newly acquired German skills into practice do it right: in German the word "Popo" (capitalized in German as it is a noun) is rather a children's term for buttocks, it cannot meaningfully be used anymore when the size of the object in question has grown across a certain subjective threshold.

If nobody feels very strongly about that, I would just stay with the short term explicitly used by the RFC for the relevant ASN.1 field identifier.o

Reply:
yeah, as far as the ASN.1 terms are used, its probably okay.
Honestly I had to laugh and could not stop for a while :) when I read the create_popo thing...